File: CustomContactInvite/src/main/java/com/verint/xm/demo/customcontactinvite/MainActivity.java
package com.verint.xm.demo.customcontactinvite;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import com.verint.xm.sdk.Core;

public class MainActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        setupToolbar(false);
    }

    public void launchCustomInvite1Activity(View view) {
        startActivity(new Intent(this, CustomInvite1Activity.class));
    }

    public void launchCustomInvite2Activity(View view) {
        startActivity(new Intent(this, CustomInvite2Activity.class));
    }

    public void resetState(View view) {
        // Reset the state of the ForeSee SDK. So that we may be eligible for a new invite
        // based on the criteria in exp_configuration.json
        Core.resetState();
    }
}

File: RulesEngineSample/src/main/java/com/verint/xm/advancedsample/MainActivity.kt
package com.verint.xm.rulesenginesample

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.verint.xm.sdk.Core
import com.verint.xm.sdk.StoryEngine

class MainActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.main)

        setupToolbar()
    }

    fun resetCounters(@Suppress("UNUSED_PARAMETER") view: View) {
        // Reset the state of the ForeSee SDK
        Core.resetState()
    }

    fun onPageViewsClicked(@Suppress("UNUSED_PARAMETER") view: View) {
        Core.resetState()

        PageActivity.pageId = 1
        // Invite will be triggered after 3 page views
        val intent = Intent(this, PageActivity::class.java)
        startActivity(intent)
    }

    fun onLaunchCountClicked(@Suppress("UNUSED_PARAMETER") view: View) {
        Core.resetState()

        // Invite will be triggered after 5 launch count
        val intent = Intent(this, LaunchCountActivity::class.java)
        startActivity(intent)
    }

    fun onPropertyClicked(@Suppress("UNUSED_PARAMETER") view: View) {
        Core.resetState()

        // Invite will be triggered after 5 launch count
        val intent = Intent(this, PropertyActivity::class.java)
        startActivity(intent)
    }

    fun onManualSurveyClicked(@Suppress("UNUSED_PARAMETER") view: View) {
        StoryEngine.runStory("manual_survey")
    }
}
